endpoints/cookie_sync.go

```diff
@@ -125,6 +131,21 @@ func (c *cookieSyncEndpoint) parseRequest(r *http.Request) (usersync.Request, pr
 		}
 	}
 
+	if request.Account != "" {
+		accountInfo, errs := accountService.GetAccount(context.Background(), c.config, c.accountsFetcher, request.Account)
+		if len(errs) > 0 {
+			return usersync.Request{}, privacy.Policies{}, combineErrors(errs)
+		}
+		if request.Limit == 0 {
+			request.Limit = accountInfo.CookieSync.DefaultLimit
+		} else if request.Limit > accountInfo.CookieSync.MaxLimit {
+			request.Limit = accountInfo.CookieSync.MaxLimit
+		}
+		if request.CooperativeSync == nil {
+			request.CooperativeSync = &accountInfo.CookieSync.DefaultCoopSync
+		}
```

Comment:
There is a logic gap here. As a publisher, I expect my max limit to always be honored. However, if you have a 0 limit and a 0 default limit, the limit will remain at 0 which is interpreted as unlimited. I had luck with the following:

```
	if request.Account != "" {
		accountInfo, errs := accountService.GetAccount(context.Background(), c.config, c.accountsFetcher, request.Account)
		if len(errs) > 0 {
			return usersync.Request{}, privacy.Policies{}, combineErrors(errs)
		}

		if request.Limit == 0 {
			request.Limit = accountInfo.CookieSync.DefaultLimit
		}

		if request.Limit == 0 || request.Limit > accountInfo.CookieSync.MaxLimit {
			request.Limit = accountInfo.CookieSync.MaxLimit
		}

		if request.CooperativeSync == nil {
			request.CooperativeSync = &accountInfo.CookieSync.DefaultCoopSync
		}
	}
```